src/main/kotlin/org/rust/ide/intentions/AddDerive.kt

```diff
@@ -0,0 +1,62 @@
+package org.rust.ide.intentions
+
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.codeStyle.CodeStyleManager
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.appendMetaToList
+import org.rust.lang.core.psi.util.parentOfType
+
+class AddDerive : PsiElementBaseIntentionAction()
+                , IntentionAction {
+    override fun getFamilyName() = "Add derive clause"
+    override fun getText() = "Add Derive Clause"
+    override fun startInWriteAction() = true
+
+    override fun invoke(project: Project, editor: Editor?, element: PsiElement) {
+        val item = element.parentOfType<RustItem>() ?: return
+        val existingDeriveAttr = item.outerAttrList.find { it.metaItem?.identifier?.textMatches("derive") ?: false }
+
+        val deriveAttr = if (existingDeriveAttr != null) existingDeriveAttr
```

Comment:
Nope, it cannot. I attempted that at first, but apparently IDEA sees the right hand side as a closure, not as a block.